presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -830,4 +832,16 @@ public static void closeWithSuppression(RecordCursor recordCursor, Throwable thr
             }
         }
     }
+
+    public static List<HiveType> getAllStructFieldTypeInfos(HiveType hiveType)
+    {
+        return ((StructTypeInfo) hiveType.getTypeInfo()).getAllStructFieldTypeInfos().stream()
+                .map(typeInfo -> HiveType.valueOf(typeInfo.getTypeName()))
+                .collect(Collectors.toList());
```

Comment:
In general, we use `ImmutableList.toImmutableList` in Guava instead. Also, we generally statically import this function whenever we use it.